custom/icds_reports/static/js/directives/adhaar-beneficiary/adhaar-beneficiary.directive.js

```diff
@@ -3,11 +3,12 @@ var url = hqImport('hqwebapp/js/initial_page_data').reverse;
 
 function AdhaarController($scope, $routeParams, $location, $filter, demographicsService, locationsService,
     dateHelperService, navigationService, userLocationId, storageService, haveAccessToAllLocations,
-    baseControllersService, isAlertActive, isMobile) {
+    baseControllersService, isAlertActive, isMobile, haveAccessToFeatures) {
     baseControllersService.BaseController.call(this, $scope, $routeParams, $location, locationsService,
         dateHelperService, navigationService, userLocationId, storageService, haveAccessToAllLocations,
         false, isMobile);
     var vm = this;
+    vm.haveAccessToFeatures = haveAccessToFeatures;
```

Comment:
should we just put this in the base controller? it looks like basically every subclass now adds it

Reply:
I thought of it. But `haveAccessToFeatures` is explicitly set to false, in the line above. So, I had to write it in every subclass

Reply:
Oh. Do you know why that is? I think it is probably just because it wasn't being passed in and someone (likely me) was too lazy to pass it through every subclass.

I think it *should* be passed up if it already goes to the `baseController`. Doesn't that imply that it's impossible to feature-flag anything in the `baseController`? That feels more like a bug than intended functionality to me.

Also, one nice thing about this situation (I think?) is that if there is a bug that depends on it being `False` it will only affect people who have the flag enabled, which will only be internal users.

Reply:
Moved it to base controller